reaper.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+)
+
+func forkExec() {
+
+	// run supercronic in other process
+	pwd, err := os.Getwd()
+	if err != nil {
+		logrus.Fatalf("Failed to get current working directory: %s", err.Error())
+		return
+	}
+
+	pattrs := &syscall.ProcAttr{
+		Dir: pwd,
+		Env: os.Environ(),
+		Files: []uintptr{
+			uintptr(syscall.Stdin),
+			uintptr(syscall.Stdout),
+			uintptr(syscall.Stderr),
+		},
+	}
+
+	pid, err := syscall.ForkExec(os.Args[0], os.Args, pattrs)
+	if err != nil {
+		logrus.Fatalf("Failed to fork exec: %s", err.Error())
+		return
+	}
+
+	// forward signal to supercronic
+	signalToFork(pid)
+	// got supercronic exit status
+	wstatus := reapChildren(pid)
+	os.Exit(wstatus.ExitStatus())
+}
+
+func signalToFork(pid int) {
+	p, err := os.FindProcess(pid)
+	if err != nil {
+		logrus.Fatalf("Failed findProcess supercronic pid:%d,%s", pid, err.Error())
+	}
+	termChan := make(chan os.Signal, 1)
+	signal.Notify(termChan, siginalList...)
+	go func() {
+		for {
+			s := <-termChan
+			if err := p.Signal(s); err != nil {
+				logrus.Errorf("Failed to send signal to supercronic: %s", err.Error())
+			}
+		}
+	}()
+}
+
+// copy from https://github.com/ramr/go-reaper
+// modify for wait exit status of supercronic
+// without modify, supercronic exit status may not be obtained
+
+// Be a good parent - clean up behind the children.
+func reapChildren(superCrondPid int) syscall.WaitStatus {
+	var notifications = make(chan os.Signal, 1)
+
+	go sigChildHandler(notifications)
+
+	// all child
+	const pid = -1
+	var wstatus syscall.WaitStatus
+
+	for {
+		var sig = <-notifications
+		logrus.Debugf("reaper received signal %v\n", sig)
+		for {
+			pid, err := syscall.Wait4(pid, &wstatus, 0, nil)
```

Comment:
I understand this works because of the way the variables are scoped but it makes the shadowing makes it confusing to follow

```suggestion
	const rpid = -1
	var wstatus syscall.WaitStatus

	for {
		var sig = <-notifications
		logrus.Debugf("reaper received signal %v\n", sig)
		for {
			pid, err := syscall.Wait4(rpid, &wstatus, 0, nil)
```